### PR TITLE
EHPR-80 | Third step validation

### PR DESCRIPTION
**Pull request by arranfw**
- `SkillInformation` validation now deeply validates stream/spec/subspecs using the data file. 
- Renamed subSpecialty -> subspecialty everwhere, this was causing confusion
- added `employmentCircumstance` question as per design update
- Converted applicant stream data to normalized shape and added some ORM-like methods to retrieve it's data
- FE methods for populating streams updated and simplified

### CODE REVIEW

**Review by fwpushan — APPROVED** [review]
LGTM 👍

**Comment by hitankar-freshworks** [comment]
I maybe be overly cautious here but FE and BE are sharing validation. Although I am not saying this is not happening, any structural changes that improve the FE validation needs checking to make sure BE validation is updated. 

The payload sent to the API endpoint can be used and altered even though the FE validates. Any lapse in revalidation at the API level can be risky.

**Review by fwpushan — APPROVED** [review]
LGTM

File: packages/common/src/dto/is-array-of-specialties.decorator.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+};
+
+const subspecialtyNotListed = (subspecialty: SubspecialtyDTO, formSpecialties: Subspecialty[]) => {
+  if (!formSpecialties.find(formSpecialty => formSpecialty.id === subspecialty.id)) {
+    return true;
+  }
+};
+
+@ValidatorConstraint({ name: 'specialties', async: false })
+export class IsArrayOfSpecialties implements ValidatorConstraintInterface {
```
Comment (arranfw) [inline packages/common/src/dto/is-array-of-specialties.decorator.ts:43]: I'm not sure if there is a way to simplify this at all, this is where the majority of the debt from this ticket lays

**Pull request merged**